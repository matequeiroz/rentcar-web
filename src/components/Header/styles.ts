import styled from "@emotion/styled";

export const Header = styled.header`
    background-color: var(--black-900);
    padding: 0 1rem;
    font-family: 'Inter';
    font-weight: 400;
    font-style: normal;
    min-width: 375px;

`

export const Content = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`