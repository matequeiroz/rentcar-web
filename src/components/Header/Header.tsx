import Image from "next/image";

import * as S from './styles'

export default function Header() {
  return (
    <S.Header data-testid="Header-component">
      <S.Content>
        <Image src={"/images/logo.svg"} alt="RentCar Logo" width={150} height={100} />
      </S.Content>
    </S.Header>
  );
}
