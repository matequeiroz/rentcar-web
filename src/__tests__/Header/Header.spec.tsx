import { render, screen } from "@testing-library/react";

import Header  from "@/components/Header/Header";

describe("Header.tsx", () => {

  it("should render the component", () => {
    const { container } = render(<Header />);
    expect(screen.getByTestId('Header-component')).toBeInTheDocument()
    expect(screen.getByRole("img")).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  });
});
