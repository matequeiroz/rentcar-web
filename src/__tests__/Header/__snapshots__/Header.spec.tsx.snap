// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.tsx should render the component 1`] = `
<div>
  <header
    class="css-9gkpj2"
    data-testid="Header-component"
  >
    <div
      class="css-4mw0p4"
    >
      <img
        alt="RentCar Logo"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/images/logo.svg"
        style="color: transparent;"
        width="150"
      />
    </div>
  </header>
</div>
`;
